lib/pbench/server/cache_manager.py

```diff
@@ -767,56 +630,46 @@ def __getitem__(self, dataset_id: str) -> Tarball:
             raise TarballNotFound(dataset_id) from None
 
     def _clean_empties(self, controller: str):
-        """
-        Remove empty controller directories from the RESULTS and INCOMING
-        trees. If there are no remaining tarballs in the ARCHIVE controller
-        directory, remove all empty state subdirectories; and, if the
-        controller directory is now empty remove it as well.
+        """Remove empty controller directories from the ARCHIVE tree.
 
         Args:
             controller: Name of the controller to clean up
         """
-        results = self.options.RESULTS / controller
-        self.delete_if_empty(results)
-        incoming = self.options.INCOMING / controller
-        self.delete_if_empty(incoming)
         archive = self.options.ARCHIVE / controller
         if archive.exists() and not any(archive.glob(f"*{Dataset.TARBALL_SUFFIX}")):
             self.delete_if_empty(archive)
             del self.controllers[controller]
 
-    def _add_controller(self, directory: Path) -> None:
-        """
-        Create a new Controller object, add it to the set of known controllers,
-        and append the discovered datasets (tarballs) to the list of known
-        datasets (by internal dataset ID) and tarballs (by tarball base name).
+    def _add_controller(self, directory: Path, cache: Path) -> None:
+        """Create a new Controller object
+
+        Add a new controller to the set of known controllers, and append the
+        discovered datasets (tarballs) to the list of known datasets (by
+        internal dataset ID) and tarballs (by tarball base name).
 
         Args:
             directory: A controller directory within the ARCHIVE tree
         """
```

Comment:
The docstring is missing the description of the `cache` parameter...but, since that parameter doesn't seem to be used by this function, perhaps it should be removed instead of fixing the docstring?  (Both calls supply the value already being used at line 653.)

Reply:
Indeed...